src/main/java/io/jenkins/plugins/checks/github/CheckRunGHEventSubscriber.java

```diff
@@ -26,77 +30,150 @@
+            return gitHubSCMFacade.findGitHubSCMSource((Job<?, ?>)item).isPresent();
         }
 
         return false;
     }
 
     @Override
     protected Set<GHEvent> events() {
-        return Collections.unmodifiableSet(new HashSet<>(Arrays.asList(GHEvent.CHECK_RUN)));
+        return Collections.unmodifiableSet(new HashSet<>(Collections.singletonList(GHEvent.CHECK_RUN)));
     }
 
     @Override
     protected void onEvent(final GHSubscriberEvent event) {
          // TODO: open a PR in Checks API to expose properties in GHRequestedAction
-        JSONObject json = JSONObject.fromObject(event.getPayload());
+        final String payload = event.getPayload();
+        JSONObject json = JSONObject.fromObject(payload);
         if (!json.getString("action").equals("requested_action")
                 || !json.getJSONObject("requested_action").get("identifier").equals("rerun")) {
+            LOGGER.log(Level.FINE, "Unsupported check run event: " + payload.replaceAll("[\r\n]", ""));
             return;
         }
 
-        GHEventPayload.CheckRun payload = null;
+        GHEventPayload.CheckRun checkRun;
         try {
-            payload = GitHub.offline().parseEventPayload(new StringReader(event.getPayload()), GHEventPayload.CheckRun.class);
+            checkRun = GitHub.offline().parseEventPayload(new StringReader(payload), GHEventPayload.CheckRun.class);
         }
         catch (IOException e) {
-            LOGGER.log(Level.WARNING,
-                    ("Received malformed CheckRun event: " + event.getPayload()).replaceAll("[\r\n]", ""), e);
-            return;
+            throw new IllegalStateException("Could not parse check run event: " + payload.replaceAll("\r\n", ""), e);
         }
 
-        final GHRepository repository = payload.getRepository();
-        final String repoUrl = repository.getHtmlUrl().toExternalForm();
-        if (!REPOSITORY_URL_PATTERN.matcher(repoUrl).matches()) {
-            LOGGER.log(Level.WARNING, ("Malformed repository URL: " + repoUrl).replaceAll("[\r\n]", ""));
-            return;
+        LOGGER.log(Level.INFO, "Received rerun request through GitHub checks API.");
+        try (ACLContext ignored = ACL.as(ACL.SYSTEM)) {
+            scheduleRerun(checkRun, payload);
         }
+    }
 
-        ACL.impersonate(ACL.SYSTEM, () -> {
-            final JSONArray pullRequests = json.getJSONObject("check_run").getJSONArray("pull_requests");
-            String branchName = "master";
-            if (pullRequests.size() != 0) {
-                branchName = "PR-" + pullRequests.getJSONObject(0).get("number");
-            }
+    private void scheduleRerun(final GHEventPayload.CheckRun checkRun, final String payload) {
+        final GHRepository repository = checkRun.getRepository();
+        final String branchName = getBranchName(JSONObject.fromObject(payload));
+
+        for (Job<?, ?> job : jenkinsFacade.getAllJobs()) {
```

Comment:
ouch this seems like an expensive way of doing it, but I guess there's no better way

Reply:
yes, the plugins I saw are doing this way, iterating all the jobs...